spec/spec_helper.rb

```diff
@@ -36,3 +36,62 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 end
+
+def stub_fastly_get_service(service_id)
+  requests = []
+
+  # Fastly#get_service. Return a service with two VCL "versions" (https://docs.fastly.com/api/config#version)
+  requests << stub_request(:get, "https://api.fastly.com/service/#{service_id}")
+    .to_return(body: File.read("spec/fixtures/fastly-get-service-response.json"))
+
+  # We clone the latest active VCL version, which returns the latest version
+  requests << stub_request(:put, "https://api.fastly.com/service/#{service_id}/version/2/clone")
+    .to_return(body: File.read("spec/fixtures/fastly-put-clone.json"))
+end
+
+def stub_fastly_apply_service_changes(service_id, is_bouncer: false)
```

Comment:
I like the `stub_fastly_get_service` helper above! I'm less thrilled by the `stub_fastly_apply_service_changes` method - it does a lot, and the `is_bouncer` boolean adds complexity too.

I think it would be easier to maintain if it were split into multiple methods, and instead of making the helper methods know what Bouncer is, we just pass the service types directly. Something like:

```ruby
let(:service_types) { %w[healthcheck cache_settings request_settings response_object header gzip] }

it "deploys the VCL for bouncer" do
  service_types << %w[backend condition]
  @requests = stub_fastly_delete_old_vcl(service_id, service_types)
  @requests += stub_fastly_activate_new_vcl(service_id)
```

Does that make sense?

Reply:
Yes, this is a good point.